sleap/nn/tracking.py

```diff
@@ -311,6 +318,119 @@ def flow_shift_instances(
         return shifted_instances
 
 
+@attr.s(auto_attribs=True)
+class FlowMaxTracksCandidateMaker(FlowCandidateMaker):
+    """Class for producing optical flow shift matching candidates with a maximum on the number of tracks.
+
+    Attributes:
+        min_points: Minimum number of points that must be detected in the new frame in
+            order to generate a new shifted instance.
+        img_scale: Factor to scale the images by when computing optical flow. Decrease
+            this to increase performance at the cost of finer accuracy. Sometimes
+            decreasing the image scale can improve performance with fast movements.
+        of_window_size: Optical flow window size to consider at each pyramid scale
+            level.
+        of_max_levels: Number of pyramid scale levels to consider. This is different
+            from the scale parameter, which determines the initial image scaling.
+        save_shifted_instances: If True, save the shifted instances between elapsed
+            frames.
+        track_window: How many frames back to look for candidate instances to match
+            instances in the current frame against.
+
+    """
+
+    max_tracks: int = None
+    min_points: int = 0
+    img_scale: float = 1.0
+    of_window_size: int = 21
+    of_max_levels: int = 3
+    save_shifted_instances: bool = False
+    track_window: int = 5
+
+    shifted_instances: Dict[
+        Tuple[int, int], List[ShiftedInstance]  # keyed by (src_t, dst_t)
+    ] = attr.ib(factory=dict)
+
+    @property
+    def uses_image(self):
+        return True
+
+    def get_candidates(
+        self,
+        track_matching_queue_dict: dict(),
+        t: int,
+        img: np.ndarray,
+        *args,
+        **kwargs,
+    ) -> List[ShiftedInstance]:
+        candidate_instances = []
+
+        # Prune old shifted instances to save time and memory
+        self.prune_shifted_instances(t)
+        number_of_tracks = 0
+        # -> List[ShiftedInstance]
+        def get_ref_instances(r_t, r_img) -> List[InstanceType]:
+            instances = []
+            for track, matched_items in track_matching_queue_dict.items():
+                instances + [
```

Comment:
```suggestion
                instances += [
```
Add test where this fails before adding fix